src/main/java/com/allocadia/carbonite/PersistedObjectCache.java

```diff
@@ -20,4 +24,14 @@
     public <T, I> T getObjectById(Class<T> clazz, I id, Function<I, T> fetcher) {
         return getInstanceCache(clazz, id).computeIfAbsent(Objects.requireNonNull(id), fetcher.andThen(Objects::requireNonNull));
     }
+
+    public <T extends PersistedObject> void addDirty(T po) {
```

Comment:
I was debating adding the `po.setDirty(field)` to this call as well (and adding a `String field` param). That way the aspect wouldn't have any logic in it and only provides a (nice compile time) method of setting the dirty state.

But is this object public? Is it ok having this on the public class?

WDTY?